InnerEye/ML/Histopathology/models/encoders.py

```diff
@@ -140,4 +140,6 @@ class HistoSSLEncoder(TileEncoder):
     def _get_encoder(self) -> Tuple[Callable, int]:
         resnet18_model = resnet18(pretrained=False)
         histossl_encoder = load_weights_to_model(self.WEIGHTS_URL, resnet18_model)
-        return setup_feature_extractor(histossl_encoder, self.input_dim)  # type: ignore
+        histossl_encoder.fc = torch.nn.Sequential()
+        num_features = 512
```

Comment:
To avoid hard-coding `num_features`, one option is to take `resnet18_model.fc.in_features`

Reply:
Thanks, done